mne/io/persyst/persyst.py

```diff
@@ -0,0 +1,407 @@
+        for t_idx, (_description, (_onset, _duration)) in \
+                enumerate(comments_dict.items()):
+            # extract the onset, duration, description to
+            # create an Annotations object
+            onset[t_idx] = _onset
+            duration[t_idx] = _duration
+            description[t_idx] = _description
+        annot = Annotations(onset, duration, description)
+        self.set_annotations(annot)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a segment of data from a file.
+
+        The Persyst software  records raw data in either 16 or 32 bit
+        binary files. In addition, it stores the calibration to convert
+        data to uV in the lay file.
+        """
+        dtype = self._raw_extras[fi]['dtype']
+        n_chs = self._raw_extras[fi]['n_chs']
+        dat_fname = self._filenames[fi]
+
+        # compute samples count based on start and stop
+        time_length_samps = stop - start
+
+        # read data from .dat file into array of correct size, then calibrate
+        # records = recnum rows x inf columns
+        if time_length_samps == -1:
+            count = -1  # elements of size precision to read
+        else:
+            count = time_length_samps * n_chs
+
+        # seek the dat file
+        with open(dat_fname, 'rb') as dat_file_ID:
+            # allow offset to occur
+            dat_file_ID.seek(n_chs * dtype.itemsize * start, 1)
+
+            # read in the actual record starting at possibly offset
+            record = np.fromfile(dat_file_ID, dtype=dtype,
+                                 count=count)
+
+        # chs * rows
+        # cast as float32; more than enough precision
+        record = np.reshape(record, (n_chs, -1), 'F').astype(np.float32)
+
+        # calibrate to convert to V and handle mult
+        _mult_cal_one(data, record, idx, cals, mult)
+
+
+def _get_subjectinfo(patient_dict):
+    # attempt to parse out the birthdate, but if it doesn't
+    # meet spec, then it will set to None
+    birthdate = patient_dict.get('birthdate')
+    if '/' in birthdate:
+        try:
+            birthdate = datetime.strptime(birthdate, '%m/%d/%y')
+        except ValueError:
+            birthdate = None
+            print('Unable to process birthdate of %s ' % birthdate)
+    elif '-' in birthdate:
+        try:
+            birthdate = datetime.strptime(birthdate, '%d-%m-%y')
+        except ValueError:
+            birthdate = None
+            print('Unable to process birthdate of %s ' % birthdate)
+
+    subject_info = {
+        'first_name': patient_dict.get('first'),
+        'middle_name': patient_dict.get('middle'),
+        'last_name': patient_dict.get('last'),
+        'sex': patient_dict.get('sex'),
+        'hand': patient_dict.get('hand'),
+        'his_id': patient_dict.get('id'),
+        'birthday': birthdate,
+    }
+
+    # Recode sex values
+    if subject_info['sex'] in ('m', 'male'):
```

Comment:
I would use a dict lookup to avoid all of these uncovered lines and make it easier to expand later:
```
sex_dict = dict(
    m=...,
    male=...,
    f=...,
    female=...,
)
subject_info['sex'] = sex_dict.get(subject_info['sex'], FIFF.FIFFV_SEX_UNKNOWN)
```